mmdet/core/visualization/image.py

```diff
@@ -115,43 +289,65 @@ def imshow_det_bboxes(img,
     ax = plt.gca()
     ax.axis('off')
 
-    polygons = []
-    color = []
-    for i, (bbox, label) in enumerate(zip(bboxes, labels)):
-        bbox_int = bbox.astype(np.int32)
-        poly = [[bbox_int[0], bbox_int[1]], [bbox_int[0], bbox_int[3]],
-                [bbox_int[2], bbox_int[3]], [bbox_int[2], bbox_int[1]]]
-        np_poly = np.array(poly).reshape((4, 2))
-        polygons.append(Polygon(np_poly))
-        color.append(bbox_color[label])
-        label_text = class_names[
-            label] if class_names is not None else f'class {label}'
-        if len(bbox) > 4:
-            label_text += f'|{bbox[-1]:.02f}'
-        ax.text(
-            bbox_int[0],
-            bbox_int[1],
-            f'{label_text}',
-            bbox={
-                'facecolor': 'black',
-                'alpha': 0.8,
-                'pad': 0.7,
-                'edgecolor': 'none'
-            },
-            color=text_color[label],
-            fontsize=font_size,
-            verticalalignment='top',
-            horizontalalignment='left')
-        if segms is not None:
-            color_mask = mask_color[labels[i]]
-            mask = segms[i].astype(bool)
-            img[mask] = img[mask] * 0.5 + color_mask * 0.5
+    max_label = int(max(labels) if len(labels) > 0 else 0)
+    if segms is not None:
+        mask_color = get_palette(mask_color, max_label + 1)
+        colors = [mask_color[label] for label in labels]
+        colors = np.array(colors, dtype=np.uint8)
+        draw_masks(ax, img, segms, colors, with_edge=True)
+
+    num_bboxes = 0
+    if bboxes is not None:
+        num_bboxes = bboxes.shape[0]
+        bbox_color = palette_val(get_palette(bbox_color, max_label + 1))
+        _labels = labels[:num_bboxes]
+        colors = [bbox_color[label] for label in _labels]
+        draw_bboxes(ax, bboxes, colors, alpha=0.8, thickness=thickness)
 
-    plt.imshow(img)
+    text_color = palette_val(get_palette(text_color, max_label + 1))
+    colors = [text_color[label] for label in labels]
+
+    if bboxes is not None:
```

Comment:
It seems that lines 307&308 can be put before line 292, then the lines 310-325 can be processed under one `if bboxes is not None`. Try to do a similar thing for segmes if segms is not None.

Reply:
Good suggestion！